tests/cloudpickle_test.py

```diff
@@ -677,34 +677,35 @@ def __init__(self, x):
 
         self.assertEqual(set(weakset), set([depickled1, depickled2]))
 
-    def test_ignoring_whichmodule_exception(self):
-        class FakeModule(object):
-            def __getattr__(self, name):
-                # This throws an exception while looking up within
-                # pickle.whichimodule.
-                raise Exception()
+    def test_faulty_module(self):
+        for module_name in ['_faulty_module', '_missing_module', None]:
```

Comment:
I cannot use `@pytest.mark.parametrize` because we derive from `unittest.TestCase`.

Reply:
Maybe new tests could be added in a `pytest` style? (i.e.: as separate functions instead of `unittest.TestCase` methods) This would result in a bit less boilerplate, less indentation and would allow the use of `assert` instead of `assertEqual()`.

Eventually we could rewrite all tests with this style if you all like the idea. Should not be too hard. Willing to help with that. :blush:

Reply:
We still need to use a base class for the various protocols (but not necessarily inheriting the `unittest.TestCase` class).

Otherwise one would have to parametrize all the test with the 2 protocols.

Anyway I think it's fine like this for now.